### DISCUSSION TITLE
技术文档自动生成工具

### DISCUSSION THREAD

**Discussion by arvinxx** [lobehub/.github#8] [Ideas]
> 程序员最痛恨的两件事：**写文档** 与 **别人不写文档**

这个 idea 会来讨论我们是否有可能直接基于源码，生成一份生产可用的文档。

## 内容生成

一个 npm 包完整的文档一般会包括以下部分：
1. 简介：告诉用户这个库是干什么的；
2. 快速上手： 如何在短短几分钟之内上手使用，让用户快速体验到 aha 时刻；
3. 详细介绍：针对更加深入的使用场景，提供完整的使用指南，一般以使用场景进行聚合；
4. API 参考：根据每一个方法，提供相应的 api 说明文档，以供用户详细了解 api 的每个参数的用法。

完全还原 API 的文档，生成可以使用 typedoc ，不过它的使用感官体验往往欠佳，这个可能不好解决。

<img width="1829" alt="image" src="https://github.com/lobehub/.github/assets/28616219/510cf6e0-cade-4d8a-a5c7-834d35b32fe9">

一般的 API 参考，在 dumi 中是使用文档解决的。因此我们需要研究下是否可能可以通过 ai + 源码，自动相应的介绍文档。

然后关于简介、快速上手、详细介绍，可能需要看看是否有可能通过 ai 生成。

## 国际化

基于一篇中文/英文，产出目标语种的 md。这个基于 lobe-i18n 应该就可以实现。

**Comment by arvinxx** [comment]
希望有的指令是:

## 1. 生成

```shell
$ lobe-docs gen src/core -o docs/api
```

从 `src/core` 下识别代码，生成文档，并将其放到 `docs/api`  目录下

由于不同类型的文档，结构可能不一样，因此生成的配置可能也需要指定：

- `readme`: Readme 文档
- `api`: API 介绍
- `usage`: 使用指南
- `intro`: 快速上手

## 2. 翻译
```shell
$ lobe-docs i18n docs/api
```

自动识别 `docs/api` 下的 md 文档，翻译为目标语言

**Comment by canisminor1990** [comment]
就是增量难做

**Reply by arvinxx** [reply]
不做增量。头大的往往是首次。后面改动的一般不多

**Comment by xihajun** [comment]
https://nn.labml.ai/diffusion/stable_diffusion/model/unet.html

这种比较好做
